src/game/MotionGenerators/TargetedMovementGenerator.cpp

```diff
@@ -494,6 +494,7 @@ bool ChaseMovementGenerator::DispatchSplineToPosition(Unit& owner, float x, floa
         this->i_path = new PathFinder(&owner);
 
     bool gen = false;
+    z += owner.GetHoverOffset();
```

Comment:
Ok so, this is kinda gonna fuck you. The way I understand hover, mob normally moves on ground, but is actually displayed X over ground. So pathfinding should pathfind on the floor and then just offset the points. That is doable in UpdateAllowedPositionZ for example and should solve your problems. Check how tc does it, logically our function works the same just with a couple diff calls. Also, should prolly remove hover flag from IsFlying in wotlk only. In tbc hover is actually used for flight.

Reply:
in sniffs at least the waypoint positions are actually above ground

Reply:
Yes, absolutely, but the pathfinding occurs on the ground. Because hover is not flying, hover is copying floor at Z offset.

Reply:
True. I don't know enough about the movement system to add the offset to the proper place, then

Reply:
![image](https://github.com/cmangos/mangos-wotlk/assets/7995382/02b6283a-544b-4d41-9957-e331784f1662)